src/libraries/System.Net.WebSockets.Client/src/System/Net/WebSockets/BrowserWebSockets/BrowserWebSocket.cs

```diff
@@ -318,23 +323,27 @@ private void NativeCleanup()
 
         public override void Dispose()
         {
-            int priorState = Interlocked.Exchange(ref _state, (int)InternalState.Disposed);
-            if (priorState == (int)InternalState.Disposed)
+            if (!_disposed)
             {
-                // No cleanup required.
-                return;
-            }
+                if (_state < (int)InternalState.Aborted) {
+                    Interlocked.Exchange(ref _state, (int)InternalState.Disposed);
```

Comment:
I'm curious about the Interlocked. If relevant code can be running on multiple threads concurrently, then what's to prevent the _state < Aborted being invalid the moment after it's checked?  And if relevant code can't be running on multiple threads concurrently, why the interlocked?

Reply:
Will take a look at these.  Thanks for the heads up.

Reply:
Addressed in recent commits